lib/common.js

```diff
@@ -344,6 +344,34 @@ function tokenize(str = '') {
   return str;
 }
 
+/**
+ * Checks if the mapping has, or inherits from a mapping that has, a no profile (NP)
+ * flag in the content profile file.
+ * @param {Object} map - The map that is being checked for a NP flag.
+ * @param {Object} contentProfiles - The contentProfiles object from the _specs.
+ */
+function hasContentProfileNoProfile(map, contentProfiles) {
+  let cp;
+  // Get content profile of map, or parent of map
+  if (map == null) {
+    return false;
+  }
+  else if (map.inheritance === 'inherited') {
+    cp = contentProfiles.findByIdentifier(map.inheritedFrom);
+  } else {
+    cp = contentProfiles.findByIdentifier(map.identifier);
+  }
+  // If contentProfile exists, iterate through rules and check for no profile
```

Comment:
The below block can be made a bit cleaner by using some fun JS!
```
return cp.rules.some(rule => rule.noProfile);
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
Oh nice, thanks for showing me that. Made the change.